### PR TITLE
Big cleanup of `glue_convert_examples_to_features`

### PR DESCRIPTION
**Pull request by julien-c**

### CODE REVIEW

File: src/transformers/data/processors/glue.py [hunk 1]
```diff
@@ -79,44 +74,15 @@ def glue_convert_examples_to_features(
...
-            token_type_ids = token_type_ids + ([pad_token_segment_id] * padding_length)
-
-        assert len(input_ids) == max_length, "Error with input length {} vs {}".format(len(input_ids), max_length)
-        assert len(attention_mask) == max_length, "Error with input length {} vs {}".format(
-            len(attention_mask), max_length
-        )
-        assert len(token_type_ids) == max_length, "Error with input length {} vs {}".format(
-            len(token_type_ids), max_length
-        )
+        inputs = tokenizer.encode_plus(example.text_a, example.text_b, max_length=max_length, pad_to_max_length=True)
```
Comment (julien-c) [inline src/transformers/data/processors/glue.py:85]: ^^ lol

Comment (julien-c) [reply src/transformers/data/processors/glue.py:85]: (and next time obviously is to move to batch_encode_plus :)

**Review by sshleifer — APPROVED** [review]
LGTM (I have limited context)

**Review by LysandreJik — APPROVED** [review]
LGTM

File: src/transformers/data/processors/glue.py [hunk 2]
```diff
@@ -73,35 +76,44 @@ def glue_convert_examples_to_features(
...
+    for i in range(len(examples)):
+        inputs = {}
+        if all_input_ids is not None:
+            inputs["input_ids"] = all_input_ids[i]
+        if all_attention_mask is not None:
+            inputs["attention_mask"] = all_attention_mask[i]
+        if all_token_type_ids is not None:
+            inputs["token_type_ids"] = all_token_type_ids[i]
+
+        feature = InputFeatures(**inputs, label=labels[i])
```
Comment (julien-c) [inline src/transformers/data/processors/glue.py:107]: This is the part that's not very pretty, @mfuntowicz – am I missing something more elegant?

**Review by thomwolf — APPROVED** [review]
LGTM if you have done extensive regression tests to be sure you are not introducing changes in the processing.

I would basically process the whole GLUE datasets with the previous and new version and flag any difference.

I think the next step is to get totally ride of `glue_convert_examples_to_features` but this could be another PR.

**Review by mfuntowicz — APPROVED** [review]

**Review by n1t0 — APPROVED** [review]

**Comment by julien-c** [comment]
There is a combinatorial explosion between `tokenizer classes` x `glue datasets` so it's not super practical to test everything, but:
- I've tested "roberta-large", "bert-base-uncased", "xlnet-based" tokenizers x "sst2" and "mnli" datasets and results are identical ✅ 
- encoding a batch of sentences or sentence pairs, while padding to a specific length, is really a native feature of a tokenizer at this point, so [those lines](https://github.com/huggingface/transformers/pull/3688/files#diff-8bc8284670454c05520b097dd51ad787R137-R139) in the current PR call the canonical API to do that. If there's a discrepancy with the historical way of tokenizing at this point it's probably outside the scope of this PR.

**Comment by julien-c** [comment]
**Note, however**, that following this PR the performance boost associated with using a fast tokenizer coupled with using `batch_encode` seems very variable, cc: @n1t0 @mfuntowicz 

On my (mac OS) local machine it takes pretty much the same time using a fast and a non-fast tokenizer (even though the fast one burns all my CPU cores).

On a Colab notebook seems like perf varies a lot between executions (https://colab.research.google.com/drive/1DXOegSz7Tyr7MeSHYBBg40kiDhu4-JPr?authuser=1#scrollTo=NlygQfeyg-5b), with the fast tokenizer not always being faster than the other one.

See [notebook](https://colab.research.google.com/drive/1DXOegSz7Tyr7MeSHYBBg40kiDhu4-JPr)

Would be interesting to dive in and do a more systematic benchmark than I did, considering that GLUE is a good benchmark for a real-world training workload.
